docs/DeveloperGuide.md

```diff
@@ -1465,3 +1465,114 @@ applications by enabling users to toggle between app states with ease.
+  serialised JSON format as we wish for the keys to be dynamically assigned at every 
+  run of the application. This way, it is unlikely for integer overflow in the Id 
+  value.
+
+* **SortCommand**: The `SortCommand` class will be changed to behave differently 
+  whenever the user enters `sort default` into the CommandBox. The `SortCommandParser` 
+  will see that none of the prefixes are present in the input. After which, it will 
+  check the preamble. If the preamble contains the string "default", the SortCommand 
+  will execute the sort logic to sort by the `Id` class.
+
+**Expected Benefits**:
+
+* **Enhanced Quality of Life**: Users do not have to restart the application just to see
+  their list in the default sorted order.
+* **Improved Usability for New Users**: By providing a way for users to revert their sorted
+  list to the default, it allows them to easily look at the latest entries that they have 
+  added.
+
+### Edit Notes
+
+**Current Implementation**: The current version of our application does not
+support the feature of editing a particular note in the note list. The only
+operations we offer for note management is adding and deleting notes as those
+two are most crucial.
+
+**Planned Enhancement**: We plan to make further enhancements to facilitate
+editing of notes. This feature will be useful for users as it saves them
+the hassle of deleting a note and re-adding it to keep their notes updated.
+
+**Proposed Changes**:
+
+* **New NoteEditCommand Class**: Create a new class called `NoteEditCommand`.
+  Similar to the `NoteInsertCommand` and `NoteDeleteCommand`, the 
+  `NoteEditCommand` will inherit from the `NoteCommand` class and behave in
+  accordance to edit a particular note. The class will require 3 parameters
+  in the constructor: The index of the application, the index of the note, 
+  and the String input note that the user wants to store.
+
+* **Additional function in InternApplication Class**: An additional function
+  is required to represent the behavior that we want in note editing. Namely,
+  `InternApplication#editNote()` will take in 2 parameters: The index of the
+  note, and the string of input note to be stored. The function makes a copy
+  of its immutable list of notes into a mutable one and swaps the `Note` in the 
+  specified index input with a newly constructed `Note` from the input string.
+  The function then returns a new `InternApplication` with the updated notes.
+
+* **NoteCommandParser Changes**: The current version of our app throws an 
+  exception when both `PREFIX_NOTE_DELETE` and `PREFIX_NOTE_INSERT` is detected.
+  The proposed change to make way for note editing is to remove the thrown 
+  exception line in place of a `NoteEditCommand` object.
+
+**Expected Benefits**:
+
+* **Faster note management for Users**: Editing notes will be reduced from a two
+  step process (Note deleting and then Note editing) into a one step process 
```

Comment:
```suggestion
  step process (Note deleting and then Note adding) into a one step process 
```
Do you perhaps mean Note deleting and then Note (re)adding?

Reply:
Oh yes, typo error